src/metabase/models/permissions.clj

```diff
@@ -57,7 +57,9 @@
    #"^/collection/(\d+)/$"                                ; readwrite permissions for a collection
    #"^/collection/(\d+)/read/$"                           ; read permissions for a collection
    #"^/collection/root/$"                                 ; readwrite permissions for the 'Root' Collection (things with `nil` collection_id)
-   #"^/collection/root/read/$"])                          ; read permissions for the 'Root' Collection
+   #"^/collection/root/read/$"                            ; read permissions for the 'Root' Collection
+   #"^/member/\d+/$"]                                     ; handle user memberships
```

Comment:
I think this new permission is a little too vague, it's not clear what exactly "handle user memberships" means. (If I have `/member/100/`, can I add User 100 to *any* Collection? Remove them from *any* Pulse? Make them a member of the `Admin` group?

Right now the permissions paths cover general read/write access patterns and don't really have more nuanced permissions for things that fall in between those extremes, except for table `query` permissions. At some point we should expand it to cover every possible action, but we should add new methods to the perms protocol besides `can-read?`/`can-write?` as part of the refactoring process. I think ideally checking for permissions in the endpoint would look something like

```clj
(api/check-can-remove-member a-pulse user-id)
```

For this PR I'd rather either:

*  Replace the existing pattern with the new improved pattern
    *  Add new permissions type(s)
    *  Add a `can-remove-member?` function to the model perms protocol as part of the PR
    *  Change other endpoints that do similar things.
*  Continue using same pattern used by `DELETE /api/card/:id/favorite` and other similar endpoints. 
    *  Don't add new permissions types or methods

I'd really rather not introduce a new competing pattern into the codebase